pandas/tests/window/test_rolling.py

```diff
@@ -81,8 +90,77 @@ def test_invalid_constructor(frame_or_series, w):
         c(window=2, min_periods=1, center=w)
 
 
+@pytest.mark.parametrize(
+    "window",
+    [
+        timedelta(days=3),
+        Timedelta(days=3),
+        "3D",
+        ExpandingIndexer(window_size=3),
+        ExponentialMovingWindowIndexer(window_size=3),
+        GroupbyIndexer(window_size=3),
+        VariableOffsetWindowIndexer(
+            index=date_range("2015-12-25", periods=5), offset=BusinessDay(1)
+        ),
+        VariableWindowIndexer(window_size=3),
+    ],
+)
+@pytest.mark.parametrize(
+    "func",
+    [
+        lambda df: df.rolling,
+        lambda df: df.groupby("key").rolling,
+    ],
+)
+def test_constructor_step_not_implemented(window, func, step):
+    # GH 15354
+    df = DataFrame(
+        {"value": np.arange(10), "key": np.array([1] * 5 + [2] * 5)},
+        index=date_range("2015-12-24", periods=10, freq="D"),
+    )
+    f = lambda: func(df)(window=window, step=step)
+    if step is None:
+        f()
+    else:
+        with pytest.raises(NotImplementedError, match="step not implemented"):
+            f()
+
+
+@pytest.mark.parametrize("agg", ["cov", "corr"])
+def test_constructor_step_not_implemented_for_cov_corr(agg, step):
+    # GH 15354
+    df = DataFrame(
+        {"value": np.arange(10), "key": np.array([1] * 5 + [2] * 5)},
+        index=date_range("2015-12-24", periods=10, freq="D"),
+    )
+    f = lambda: getattr(df.rolling(window=2, step=step), agg)(df)
+    if step is None:
```

Comment:
Maybe just test `if step if not None` since otherwise we have pretty good testing for cov/corr